internal/secret-storage-backend/server.go

```diff
@@ -0,0 +1,315 @@
+package secretstoragebackend
```

Comment:
> Maybe we can create dedicated Type but then “generic” Type for AWS Secret Manger will have const with aws_secretsmanager for Provider entry—it may look strange. 

This is something what I initially thought. I know that it's not perfect solution, but for now I would keep it as it is.

> Maybe we should hide it and just expose each "provider" separately.

That's good idea, however it would probably require us to expose every service on different port. Not sure if routing requests for exact copies of gRPC server would be easy to implement - as you wrote, we need to revisit the topic in the future, once we support more store providers🙂